tools/test/topology/test-playback.m4

```diff
@@ -52,10 +88,71 @@ PIPELINE_PCM_ADD(sof/pipe-TEST_PIPE_NAME-playback.m4,
 	1000, 0, 0,
 	8000, 192000, 48000)
 
+# Generalize the pipeline junction name as PIPELINE_JUNCTION.
+# e.g. TEST_PIPE_NAME=`crossover' --> PIPELINE_JUNCTION=`PIPELINE_CROSSOVER_1'
+define(PIPELINE_JUNCTION, concat(concat(`PIPELINE_', upcase(TEST_PIPE_NAME)), `_1'))
+
+ifdef(`TEST_HAS_PIPE2',
+`
+# Playback pipeline 2 on PCM 1 using max 2 channels of s32le.
+# Set 1000us deadline on core 0 with priority 0
+PIPELINE_PCM_ADD(sof/pipe-dai-endpoint.m4,
+	2, 1, 2, s32le,
+	1000, 0, 0,
+	8000, 192000, 48000)
+
+# connect pipelines together
+SectionGraph."pipe-sof-second-pipe" {
+        index "2"
+
+        lines [
+		# connect the second sink buffer
+                dapm(PIPELINE_SOURCE_2, PIPELINE_JUNCTION)
+        ]
+}
+')
+
+ifdef(`TEST_HAS_PIPE3',
+`
+# Playback pipeline 3 on PCM 2 using max 2 channels of s32le.
+# Set 1000us deadline on core 0 with priority 0
+PIPELINE_PCM_ADD(sof/pipe-dai-endpoint.m4,
+	3, 2, 2, s32le,
+	1000, 0, 0,
+	8000, 192000, 48000)
+
+# connect pipelines together
+SectionGraph."pipe-sof-third-pipe" {
+        index "3"
+
+        lines [
+		# connect the third sink buffer
+                dapm(PIPELINE_SOURCE_3, PIPELINE_JUNCTION)
+        ]
+}
+')
+
+ifdef(`TEST_HAS_PIPE4',
+`
+# Playback pipeline 4 on PCM 3 using max 2 channels of s32le.
```

Comment:
I wonder if this section could be extracted out to a macro call and the index is just subbed in, thoughts?